plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java

```diff
@@ -19,103 +19,204 @@
-    {
-        getShell().setText(UIConnectionMessages.dialog_edit_driver_edit_maven_title);
+    protected Composite createDialogArea(Composite parent) {
+        Composite composite = super.createDialogArea(parent);
+        {
+            GridData gd = new GridData(GridData.FILL_BOTH);
+            gd.widthHint = UIUtils.getFontHeight(composite.getFont()) * 30;
 
-        Composite composite = (Composite) super.createDialogArea(parent);
-        ((GridLayout)composite.getLayout()).numColumns = 2;
+            tabFolder = new TabFolder(composite, SWT.TOP | SWT.FLAT);
+            tabFolder.setLayoutData(gd);
+            if (originalArtifact == null) {
+                createRawTab(tabFolder);
+            }
+            createManualTab(tabFolder);
+        }
+        {
+            Group settingsGroup = UIUtils.createControlGroup(composite, UIConnectionMessages.dialog_edit_driver_edit_maven_settings, 1, GridData.FILL_HORIZONTAL, 0);
+
+            Button ignoreDependenciesCheckbox = UIUtils.createCheckbox(settingsGroup,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_ignore_transient_dependencies,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_load_optional_dependencies_tip,
+                originalArtifact != null && originalArtifact.isIgnoreDependencies(),
+                2);
+            ignoreDependenciesCheckbox.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    ignoreDependencies = ignoreDependenciesCheckbox.getSelection();
+                }
+            });
 
+            Button loadOptionalDependenciesCheckbox = UIUtils.createCheckbox(settingsGroup,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_load_optional_dependencies,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_load_optional_dependencies_tip,
+                originalArtifact != null && originalArtifact.isIgnoreDependencies(),
+                2);
+            loadOptionalDependenciesCheckbox.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    loadOptionalDependencies = loadOptionalDependenciesCheckbox.getSelection();
+                }
+            });
+        }
+
+        return composite;
+    }
+
+    private void parseArtifactText() {
+        try {
+            artifacts.clear();
+            if (REGEX_FOR_GRADLE.matcher(fieldText.getText()).find()) {
+                artifacts.addAll(parseGradle());
+            } else {
+                artifacts.addAll(parseMaven());
+            }
+            setStatus(false, NLS.bind(UIConnectionMessages.dialog_edit_driver_edit_maven_artifacts_count, artifacts.size()));
+        } catch (Exception e) {
+            setStatus(true, e.getMessage());
+            log.trace("Error parsing artifact!", e);
```

Comment:
This effectively eats the exception since `trace` is disabled by default. I think we need to include this exception in debug log.